app/assets/javascripts/govuk_publishing_components/components/input.js

```diff
@@ -0,0 +1,48 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function Input () { }
+
+  Input.prototype.start = function ($module) {
+    this.$module = $module[0]
+    this.input = this.$module.querySelectorAll('.gem-c-input')[0]
+
+    var extraText = " <span class='gem-c-input__password-long-text'>" + this.$module.getAttribute('data-password') + '</span>'
+    this.showPasswordText = this.$module.getAttribute('data-show') + extraText
+    this.hidePasswordText = this.$module.getAttribute('data-hide') + extraText
+    this.shownPassword = this.$module.getAttribute('data-announce-show')
+    this.hiddenPassword = this.$module.getAttribute('data-announce-hide')
+
+    var showHide = document.createElement('button')
+    showHide.className = 'gem-c-input__password-toggle gem-c-button govuk-button gem-c-button--secondary-quiet'
+    showHide.setAttribute('data-show', 'true')
+    showHide.setAttribute('aria-controls', this.input.getAttribute('id'))
+    showHide.innerHTML = this.showPasswordText
+    this.$module.insertBefore(showHide, this.input.nextSibling)
+
+    this.statusText = document.createElement('span')
+    this.statusText.classList.add('govuk-visually-hidden')
+    this.statusText.innerHTML = this.hiddenPassword
+    this.statusText.setAttribute('aria-live', 'polite')
+    this.$module.insertBefore(this.statusText, showHide.nextSibling)
+
+    var that = this
```

Comment:
I'm struggling to wrap my head around this approach. Why do we need to define a separate instance of `this` outside the event listener below? If this is a question of scope and the possibility of multiple show password buttons on one page, wouldn't this module handle that scoping?

Reply:
Good point, I was apparently being lazy. Have refactored.